include/deal.II/base/exceptions.h

```diff
@@ -1228,6 +1228,23 @@ namespace StandardExceptions
                                            dealii::ExcCudaError(cudaGetErrorString(error_code)))
 #endif
 
+#if defined(DEAL_II_WITH_CXX11) || !defined(DEAL_II_MSVC)
+/**
+ * A static assert which checks a compile-time condition. This is nothing more
+ * than a wrapper around the C++11 <code>static_assert</code> declaration, which
+ * we need to disable for non-C++11 and for Visual Studio.
+ *
+ * Just like <code>static_assert</code>, it takes two arguments, a
+ * <code>constexpr</code> condition and an error message.
+ *
+ * @ingroup Exceptions
+ * @author Karl Ljungkvist, 2017
+ */
+#define AssertStatic(...) static_assert(__VA_ARGS__)
```

Comment:
The problem is with expressions like `AssertStatic(mystruct<T,S>::value,"errormsg")`. The preprocessor doesn't handle the extra comma gracefully.